src/tests/Makefile

```diff
@@ -1,6 +1,6 @@
 # Makefile for tests - builds unit-test binaries
 
-CFLAGS+=-I../ -I. -g
```

Comment:
realpath is present on Debian 11 and OpenBSD 7.1, but not on macOS 12.6, so this'll break src/Makefile.osx's tests rule, at least for some versions of macOS.  I haven't thought of a workaround for that.  One way would be to have SRCROOT set somewhere else (likely src/Makefile for builds through configure or src/Makefile.osx when building on macOS) but that makes src/tests/Makefile less self-contained than your approach.